pkg/bundle/fs.go

```diff
@@ -0,0 +1,187 @@
+package bundle
+
+import (
+	"io/fs"
+	"path/filepath"
+
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// FS is a File System that contains a bundle.
+//
+// A bundle is a directory that contains files that may or may not be
```

Comment:
So do we want to enforce the semantics of a bundle here then? For example, should the `bundle.FS` know that the `manifests` directory has special meaning and enforce that it can't have sub directories? My thinking was that it would be up to the specific bundle types to actually implement the semantics and a `bundle.FS` is just a filesystem with kubernetes manifests.